cai_causal_graph/graph_components.py

```diff
@@ -310,6 +326,46 @@ def __hash__(self) -> int:
         """Return a hash value of the node identifier."""
         return super().__hash__()
 
+    def to_dict(self, include_meta: bool = True) -> dict:
+        """
+        Serialize a `cai_causal_graph.graph_components.TimeSeriesNode` instance to a dictionary.
+
+        :param include_meta: Whether to include meta information about the node in the dictionary. Default is `True`.
+        :return: The dictionary representation of the `cai_causal_graph.graph_components.TimeSeriesNode` instance.
+        """
+        dictionary = super().to_dict(include_meta)
+        # add the time lag and variable name to the dictionary
+        dictionary.update({TIME_LAG: self.time_lag, VARIABLE_NAME: self.variable_name})
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> TimeSeriesNode:
+        """
+        Return a `cai_causal_graph.graph_components.TimeSeriesNode` instance from a dictionary.
+
+        :param dictionary: The dictionary from which to create the time series node.
+        :return: The time series node created from the dictionary.
+        """
+        assert 'identifier' in dictionary, 'The dictionary must contain the `identifier` key.'
```

Comment:
Should we assert `'node_class'` is correct? Can do `assert dictionary.get('node_class', 'TimeSeriesNode') == 'TimeSeriesNode'`.

Reply:
Why? I may want convert a Node into a TimeSeriesNode even if the original type is different

Reply:
I agree. Just wonder if we want to warn or anything.